backend/tilavarauspalvelu/api/graphql/types/reservation/types.py

```diff
@@ -325,7 +337,111 @@ def resolve_pindora_info(root: Reservation, info: GQLInfo) -> PindoraInfoData |
         access_code_begins_at = begin - datetime.timedelta(minutes=access_code_valid_minutes_before)
         access_code_ends_at = end + datetime.timedelta(minutes=access_code_valid_minutes_after)
 
-        return PindoraInfoData(
+        return PindoraReservationInfoData(
+            access_code=response["access_code"],
+            access_code_generated_at=response["access_code_generated_at"],
+            access_code_is_active=access_code_is_active,
+            access_code_keypad_url=response["access_code_keypad_url"],
+            access_code_phone_number=response["access_code_phone_number"],
+            access_code_sms_number=response["access_code_sms_number"],
+            access_code_sms_message=response["access_code_sms_message"],
+            access_code_begins_at=access_code_begins_at,
+            access_code_ends_at=access_code_ends_at,
+        )
+
+    @staticmethod
+    def series_pindora_info(
+        series: RecurringReservation,
+        *,
+        begin: datetime.datetime,
+        end: datetime.datetime,
+        has_perms: bool,
+    ) -> PindoraReservationInfoData | None:
+        if series.allocated_time_slot is not None:
+            return ReservationNode.section_pindora_info(
```

Comment:
Should also check for application.application_round.sent_date

Reply:
Done.